staging/src/kubevirt.io/client-go/kubevirt/typed/core/v1/async.go

```diff
@@ -202,34 +203,33 @@ func RequestFromConfig(config *rest.Config, resource, name, namespace, subresour
 	return req, nil
 }
 
-// enrichError checks the response body for a k8s Status object and extracts the error from it.
-// TODO the k8s http REST client has very sophisticated handling, investigate on how we can reuse it
-func enrichError(httpErr error, resp *http.Response) error {
```

Comment:
This does not have any tests, can you add some as first commit to demonstrate this is still doing what it did?

Reply:
I've added to the first commit, so you can go `git checkout` && `go test -v ./...` to check.